src/JackMD/UpdateNotifier/task/UpdateNotifyTask.php

```diff
@@ -73,28 +71,24 @@ public function onRun(): void{
 				$api = $release["api"][0]["from"] . " - " . $release["api"][0]["to"];
 			}
 		}
-		
+
 		$this->setResult([$highestVersion, $artifactUrl, $api, $err]);
 	}
-	
-	/**
-	 * @param Server $server
-	 */
-	public function onCompletion(Server $server): void{
-		$pluginName = $this->pluginName;
-		$plugin = $server->getPluginManager()->getPlugin($pluginName);
+
+	public function onCompletion(Server $server) : void{
+		$plugin = Server::getInstance()->getPluginManager()->getPlugin($this->pluginName);
```

Comment:
This is already ready for PM4.

Reply:
But in 4.0 (last time i saw) the onCompletion() method didnt have any params?

Reply:
Yes, you just need to remove the param to support the 4.0